### ISSUE TITLE
Dependencies for maven project only resolved for cyclonedx, not for spdx/json

### ISSUE THREAD

**Issue by eversinc33** [nexB/scancode-toolkit#2854]
### Description

Dependencies/Packages from a maven pom.xml are only detected when outputting to cyclone-dx-xml, but not when outputting to json or spdx-tv or spdx-xml.

Im not sure if this is a missing feature that is only implemented in cyclonedx or a bug?

### How To Reproduce

I created a Hello-World-Style java project with the following dependency included in pom.xml

```xml
<dependencies>
  <dependency>
    <groupId>log4j</groupId>
    <artifactId>log4j</artifactId>
    <version>1.2.17</version>
  </dependency>
</dependencies>
```
Running scancode like so:

```bash
scancode -clpieu --spdx-tv ../../scancode-java.spdx.tv ../../Log4J-example-with-java --max-depth 0 --consolidate
```

Yields an SBOM which only includes "Files" and the main Package

```
# Package

PackageName: log4j_example_with_java
SPDXID: SPDXRef-001
PackageDownloadLocation: NOASSERTION
PackageVerificationCode: daf7c31358a7967f856b9a0eda8bee2f61abde16
PackageLicenseDeclared: NOASSERTION
PackageLicenseConcluded: NOASSERTION
PackageLicenseInfoFromFiles: NONE
PackageCopyrightText: NONE


# File
FileName: ./Log4J-example-with-java/.idea/compiler.xml
SPDXID: SPDXRef-6
FileChecksum: SHA1: 5deee417bc8e462aa80f979e9fb353f707df9b1f
LicenseConcluded: NOASSERTION
LicenseInfoInFile: NONE
FileCopyrightText: NONE

# ...
```

Same goes for the JSON output.

CycloneDX however correctly includes the log4j dependency:

```bash
scancode -clpieu --cyclonedx ../../scancode-java.json.cdx ../../Log4J-example-with-java --max-depth 0 --consolidate
```

```xml
<bom>
<!-- [...] --->
<dependencies>
    <dependency ref="pkg:maven/com.corp/SBOMEXample@1.0-SNAPSHOT">
      <dependency ref="pkg:maven/log4j/log4j@1.2.17"/>
    </dependency>
  </dependencies>
</bom>
```

### System configuration

5.13.0-27-generic #29-Ubuntu SMP Wed Jan 12 17:36:47 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

ScanCode version 30.1.0

ScanCode Output Format version 2.0.0

SPDX License list version 3.15

**Comment by pombredanne** [comment]
Thanks for the report. Deps reporting is being overhauled in @AyanSinhaMahapatra branch  #2691 which is under review :)

You wrote:
> Same goes for the JSON output.

The SPDX is missing the deps alright, but the JSON output which ScanCode's native AboutCode format should have the deps alright?

Also I am a bit at loss on how to represent dependencies in the SPDX output. What would be your take for this?

**Comment by eversinc33** [comment]
With JSON I was referring to spdx-json.

I would suggest to add the relationships such as CONTAINS, CONTAINED_BY etc., see page 80 here: https://spdx.dev/wp-content/uploads/sites/41/2020/08/SPDX-specification-2-2.pdf

I can not answer you about the native AboutCode format as that was not of interest to us when evaluating scancode toolkit and unfortunately I do not have the files anymore.

Looking forward to the overhaul, I will try out once more when #2691 is merged.

**Comment by pombredanne** [comment]
@eversinc33 re:

> With JSON I was referring to spdx-json.

we do not have this yet as an output format

> I would suggest to add the relationships such as CONTAINS, CONTAINED_BY etc., see page 80 here: https://spdx.dev/wp-content/uploads/sites/41/2020/08/SPDX-specification-2-2.pdf

What do you think would be the best way to map our models to these relationships?

**Comment by eversinc33** [comment]
> we do not have this yet as an output format

Right, I'm stupid, then I probably meant the AboutCode format, and thought it was SPDX-JSON. Sorry, tried too many different tools, and its hard to not mix memories in my mind :)

> What do you think would be the best way to map our models to these relationships?

I cannot give an elaborate answer on that, but what I would be expecting as a user, is at least to have them not as individual files, but as a Package. Then, if there are dependencies between packages, list them as Relationships, e.g. `Relationship: MainPackage DEPENDS_ON Log4J`.

But take that with a grain of salt, I am still learning about SBOM best practices.
